### PR TITLE
Add SaveNNMixin for fixing saving/loading of NNs

### LINKED ISSUE #1011: Fix saving/loading of NNs
### 🚀 Feature Request

Currently, saving of NNs is made using pickle. It doesn't seem really reliable and also isn't working for old NNs.

### Proposal

1. Refactor `save` and `load` in `SaveMixin`:
  * Add new method `_save_metadata` and move saving metadata into it.
  * Add new method `_load_metadata` and move loading metadata into it.
  * Add new method `_save_state` and move saving state into it.
  * Add new method `_load_state` and move loading state into it.
  * Refactor tests.
2. Create `SaveNNMixin` in `models.mixins`.
  * It shoud be inherited from `SaveMixin`.
  * It should redifine `_save_state` and `_load_state` to use `torch.save` and `torch.load` with dill module.
3. Move `SaveMixin` in models from `AbstractModel` to `PerSegmentModelMixin` and `MultiSegmentModelMixin`.
4. In `DeepBaseModel`
  * Add `SaveNNMixin` to list of classes in definition.
  * Remove redundant warning in documentation.
5. In `DeepARModel`
  * Add `SaveNNMixin` to list of classes in definition.
  * Remove redundant warning in documentation.
6. In `TFTModel`
  * Add `SaveNNMixin` to list of classes in definition.
  * Remove redundant warning in documentation.
7. Fix example in `custom_transform_and_model` notebook (it will fail) to work with `SaveMixin`.

### Test cases

1. Fix test for `SaveMixin`.
2. Write tests for `SaveNNMixin`.
3. Make sure tests for NNs and old NNs working.
4. Remove `pytest.mark` from tests for saving and loading.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Mr-Geekman**
## Before submitting (must do checklist)

- [x] Did you read the [contribution guide](https://github.com/tinkoff-ai/etna/blob/master/CONTRIBUTING.md)?
- [x] Did you update the docs? We use Numpy format for all the methods and classes.
- [x] Did you write any new necessary tests?
- [x] Did you update the [CHANGELOG](https://github.com/tinkoff-ai/etna/blob/master/CHANGELOG.md)?

## Proposed Changes
Look at #1011.

## Closing issues
Closes #1011.

### CODE REVIEW

File: tests/test_models/nn/test_tft.py [hunk 1]
```diff
@@ -181,10 +181,9 @@ def test_prediction_interval_run_infuture_warning_loss(example_tsds):
         assert {"target_0.02", "target_0.98"}.isdisjoint(segment_slice.columns)
 
 
-@pytest.mark.xfail(reason="Should be fixed in inference-v2.0", strict=True)
 def test_save_load(example_tsds):
     horizon = 3
-    model = TFTModel(max_epochs=2, learning_rate=[0.1], gpus=0, batch_size=64, loss=MAEPF())
```
Comment (martins0n) [inline tests/test_models/nn/test_tft.py:187]: Why do we change loss?

Comment (Mr-Geekman) [reply tests/test_models/nn/test_tft.py:187]: I probably made a mistake. As you can see, this loss is only imported in the previous test and wasn't imported in the whole test file. There are two ways to fix it:
* Import `MAEPF` in this test too.
* Use default loss.

I choose the second option.

**Review by martins0n — APPROVED** [review]
